deploy/docker/docker-entrypoint.sh

```diff
@@ -39,9 +39,8 @@ if [[ -z "$EMQX_HOST" ]]; then
     export EMQX_HOST
 fi
 
-if [[ -z "$EMQX_NODE_NAME" ]]; then
-    export EMQX_NODE_NAME="$EMQX_NAME@$EMQX_HOST"
-fi
+export EMQX_NODE_NAME=${EMQX_NODE__NAME:-$EMQX_NAME@$EMQX_HOST}
+echo "node.name = $EMQX_NODE_NAME" >> /opt/emqx/etc/emqx.conf
```

Comment:
```suggestion
if [[ -z "$EMQX_NODE_NAME" ]] && [[ -z "$EMQX_NODE_NAME" ]]; then
    export EMQX_NODE__NAME="$EMQX_NAME@$EMQX_HOST"
fi
```

Reply:
we should try not to touch emqx.conf as it can be read-only.

Reply:
another thing:
Currently 'EMQX_NODE_NAME' has higher precedence over 'EMQX_NODE__NAME'.
Although it's unlikely that one will set both intentionally, but changing the order might affect some current users.
This is why we should not take `EMQX_NODE__NAME` for `EMQX_NODE_NAME` in docker entrypoint.